### PR TITLE
Priority Queues guide improvements

### PR DESCRIPTION
**Pull request by michaelklishin**
Documentation improvements to go with rabbitmq/rabbitmq-server#1590.

### CODE REVIEW

File: site/priority.xml [hunk 1]
```diff
@@ -27,127 +27,172 @@ limitations under the License.
...
-      indicating the maximum priority the queue should support. For example,
-      using the Java client:
-    </p>
-
-    <pre class="sourcecode java">
+    <doc:section name="intro">
+      <p>
+        RabbitMQ has priority queue implementation in the core as of version <code>3.5.0</code>.
+        Any queue can be turned into a priority one using <a href="/queues.html#optional-arguments">optional arguments</a>.
+        The implementation supports a limited number of priorities.
```
Comment (acogoluegnes) [inline site/priority.xml]: Maybe we should mention the range for priorities, as the exact values show up only in the "Using Client-provided Optional Arguments" section, not in the policy one.

File: site/priority.xml [hunk 2]
```diff
@@ -27,127 +27,172 @@ limitations under the License.
...
+        Publishers then can publish prioritised messages using the
+        <code>priority</code> field of
+        <code>basic.properties</code>. Larger numbers indicate higher
+        priority.
+      </p>
+
+      <p>
+        Because the on-disk format for a priority queue is different,
+        <b>priority queues can only be defined by arguments</b>, not
+        policies. Queues can never change the number of priorities they
```
Comment (acogoluegnes) [inline site/priority.xml]: "not policies" is misleading as this is the first way we use to define priorities.

Comment (michaelklishin) [reply site/priority.xml]: That comment explains why policies are probably not a good idea in this case. This must be partially undone then 🤦‍♂️.

**Comment by michaelklishin** [comment]
The docs were updated to match https://github.com/rabbitmq/rabbitmq-server/commit/08168de6b0c3c4d7f6429f2c3d098e5080ff7b1a.

**Pull request merged**